### PR TITLE
#121 storing local data sources

### PR DESCRIPTION
**Pull request by kbernat**
#121 ready for merge.

### CODE REVIEW

File: cate/ds/local.py [hunk 1]
```diff
@@ -142,9 +174,16 @@ def add_pattern(self, name: str, files: Union[str, Sequence[str]]):
         self._save_data_source(data_source)
         return name
 
-    def query(self, name=None, monitor: Monitor = Monitor.NONE) -> Sequence[DataSource]:
+    def query(self, name=None, monitor: Monitor = Monitor.NONE) -> Sequence[LocalFilePatternDataSource]:
```
Comment (mzuehlke) [inline cate/ds/local.py]: Why this change ?

**Comment by kbernat** [comment]
@mzuehlke  - I've added two query method tests for datasources with and without temporal coverage.
Could you describe what would you like to see in remaining two combinations?

**Comment by kbernat** [comment]
tests are failing, I will take care of that.

**Comment by mzuehlke** [comment]
Hi Chris, sorry wasn't fast enough in writing the comment. I mixed up the methods query and open_dataset. Sorry for that. Thanks for taking care of this.
